linode_api4/objects/object_storage.py

```diff
@@ -566,3 +566,42 @@ class ObjectStorageKeys(Base):
         "limited": Property(),
         "regions": Property(unordered=True),
     }
+
+
+class ObjectStorageQuota(Base):
+    """
+    An Object Storage related quota information on your account.
+    Object Storage Quota related features are under v4beta and may not currently be available to all users.
+
+    API documentation: TBD
+    """
+
+    api_endpoint = "/object-storage/quotas/{quota_id}"
+    id_attribute = "quota_id"
+
+    properties = {
+        "quota_id": Property(identifier=True),
+        "quota_name": Property(),
+        "endpoint_type": Property(),
+        "s3_endpoint": Property(),
+        "description": Property(),
+        "quota_limit": Property(),
+        "resource_metric": Property(),
+    }
+
+    def usage(self):
+        """
+        Gets usage data for a specific ObjectStorage Quota resource you can have on your account and the current usage for that resource.
+
+        API documentation: TBD
+
+        :returns: The Object Storage Quota usage.
+        :rtype: MappedObject
+        """
+
+        result = self._client.get(
+            f"{type(self).api_endpoint}/usage",
+            model=self,
+        )
+
+        return MappedObject(**result)
```

Comment:
Assuming the schema of this response is guaranteed by the API, would it make sense to create a JSONObject class for the response here?